### DISCUSSION TITLE
Null QueryKey values

### DISCUSSION THREAD

**Discussion by cgatian** [TanStack/query#3627] [Q&A] [answered]
I've been looking closely at the cache using the dev-tools and realized there are some query entries that have `null` or `-1` in the query key for instances where the variable was not known at the time of the hook being instantiated. 

![image](https://user-images.githubusercontent.com/1752170/169349090-dc8b6cd6-b251-42e8-ba61-b3729d6085c7.png)
_Example: the `id` portion of the Profile is not known on initialization of useQuery hook. `null` is used for the key_

While these queries are disabled through the `enabled` property, I started to second-guess if we were doing it "right".

Some Questions:

- Does it make sense to create a cache entry if the query starts off disabled? Maybe it should only be created once it's been enabled?

- Any advice on how to avoid creating these queries?

- Am I overthinking this, and should just embrace that this is how the library works?

**Comment by TkDodo** [comment, marked as answer]
> Does it make sense to create a cache entry if the query starts off disabled? Maybe it should only be created once it's been enabled?

not possible because you could also pass `initialData` for example, or other settings that should be immediately available for other observers to be picked up.

also, you can indefinitely keep it as `disabled` and then call `refetch` to fetch it :)

> Am I overthinking this, and should just embrace that this is how the library works?

yes, it's not necessary to avoid creating these. I've added the `disabled` label to the devtools to flag these as, well, disabled :)

**Reply by cgatian** [reply]
Awesome, thank you for your time.
